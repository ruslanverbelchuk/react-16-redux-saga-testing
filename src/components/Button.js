import React from 'react';
import {connect} from 'react-redux';
import {fetchRatesStart} from '../redux/rate/rate.actions';

const styles = {
  width: '200px',
  height: '50px',
  display: 'block',
  margin: '50px auto',
  fontSize: '25px',
  border: '3px solid ',
};

class Button extends React.Component {
  render() {
    return (
      <button style={styles} onClick={this.props.fetchRatesStart}>
        Get Rates
      </button>
    );
  }
}

const mapDispatchToProps = {
  fetchRatesStart,
};

Button = connect(
  null,
  mapDispatchToProps,
)(Button);

export default Button;
