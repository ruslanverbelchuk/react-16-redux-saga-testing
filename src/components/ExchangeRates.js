import React from 'react';
import {connect} from 'react-redux';

const rateStyle = {
  width: '50%',
  margin: '0 auto',
  color: 'olive',
};
const errorMessage = {
  color: 'red',
};

let ExchangeRates = ({rate}) => {
  const renderList = [];
  rate &&
    Object.entries(rate.rates).forEach(([key, value]) =>
      renderList.push(
        <li key={key}>
          {key}: {value}
        </li>,
      ),
    );

  return rate ? (
    <div style={rateStyle}>
      {rate.error && <h2 style={errorMessage}>{rate.error}</h2>}
      {rate.base && <h2>BASE CURRENCY: {rate.base}</h2>}
      {rate.date && <h2>DATE: {rate.date}</h2>}
      {rate.rates && <ul>{renderList}</ul>}
    </div>
  ) : null;
};

const mapStateToProps = state => ({
  rate: state.rateItems,
});

ExchangeRates = connect(
  mapStateToProps,
  null,
)(ExchangeRates);

export default ExchangeRates;
