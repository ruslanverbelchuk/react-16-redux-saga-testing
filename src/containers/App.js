import React from 'react';
import Button from '../components/Button';
import ExchangeRates from '../components/ExchangeRates';
import Loading from '../components/Loading';

const App = () => (
  <div>
    <Button />
    <Loading />
    <ExchangeRates />
  </div>
);

export default App;
