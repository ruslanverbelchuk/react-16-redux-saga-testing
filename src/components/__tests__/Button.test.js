import React from 'react';
import { shallow } from 'enzyme';
import ExchangeRates from '../ExchangeRates';
import Button from '../Button';
import App from '../../containers/App';

describe('Button component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });
/*  it('has an ExchangeRate', () => {
    wrapper.find(Button).simulate('click');
    wrapper.update();
  });*/

  it('should render Button component', () => {
    expect(wrapper.find(Button)).toMatchSnapshot();
  });
});
