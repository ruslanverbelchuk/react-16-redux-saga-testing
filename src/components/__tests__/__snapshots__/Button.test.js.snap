// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render Button component 1`] = `ShallowWrapper {}`;
